src/api/telemetry/TelemetryAPI.js

```diff
@@ -298,6 +312,61 @@ export default class TelemetryAPI {
         }
     }
 
+    /**
+     * Subscribe to staleness updates for a specific domain object.
+     * The callback will be called whenever staleness changes.
+     *
+     * @method subscribeToStaleness
+     * @memberof module:openmct.TelemetryAPI~StalenessProvider#
+     * @param {module:openmct.DomainObject} domainObject the object
+     *          to watch for staleness updates
+     * @param {Function} callback the callback to invoke with staleness
+     *          updates
+     * @returns {Function} a function which may be called to terminate
+     *          the subscription to staleness updates
+     */
+    subscribeToStaleness(domainObject, callback) {
+        const provider = this.#findStalenessProvider(domainObject);
+
+        if (!this.stalenessSubscriberCache) {
+            this.stalenessSubscriberCache = {};
+        }
+
+        const keyString = objectUtils.makeKeyString(domainObject.identifier);
+        let stalenessSubscriber = this.stalenessSubscriberCache[keyString];
+
+        if (!stalenessSubscriber) {
+            stalenessSubscriber = this.stalenessSubscriberCache[keyString] = {
+                isStale: false,
```

Comment:
I am think about the case where a telemetry point is already stale before any subscriptions happen. The code currently only conveys staleness through a subscription, so there is an implicit requirement on staleness providers to immediately invoke a subscription callback with the current staleness state.

We could either document this implicit requirement, but my preference is code over documentation. So how about we add an `async isStale()` function to the staleness provider interface and to the Telemetry API? This function will always return the current staleness state, and `subscribeToStaleness` will just provide updates when it changes.

Thoughts?

Reply:
I was wondering how to handle that. I had actually added an `isStale` property to the `stalenessSubscriberCache` for each item and I update it whenever we receive updates from the provider, then if any new subscriptions come in, I immediately call the callback with the current value, BUT for the initial subscription, this could be the way to handle it.